pkg/xds/ingress/dataplane.go

```diff
@@ -0,0 +1,58 @@
+package ingress
+
+import (
+	"context"
+	"reflect"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	core_mesh "github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	"github.com/Kong/kuma/pkg/core/resources/manager"
+	"github.com/Kong/kuma/pkg/core/resources/store"
+)
+
+type ingressSet []*mesh_proto.Dataplane_Networking_Ingress
+
+func (set ingressSet) getBy(tags map[string]string) *mesh_proto.Dataplane_Networking_Ingress {
+	for _, in := range set {
+		if reflect.DeepEqual(in.Tags, tags) {
+			return in
+		}
+	}
+	return nil
+}
+
+func GetIngressByDataplanes(others []*core_mesh.DataplaneResource) []*mesh_proto.Dataplane_Networking_Ingress {
+	ingresses := make([]*mesh_proto.Dataplane_Networking_Ingress, 0, len(others))
+	for _, dp := range others {
+		if dp.Spec.GetNetworking().GetIngress() != nil {
+			continue
+		}
+		for _, dpInbound := range dp.Spec.GetNetworking().GetInbound() {
+			if dup := ingressSet(ingresses).getBy(dpInbound.GetTags()); dup != nil {
+				continue
+			}
+			ingresses = append(ingresses, &mesh_proto.Dataplane_Networking_Ingress{
+				Service: dpInbound.Tags[mesh_proto.ServiceTag],
+				Tags:    mesh_proto.SingleValueTagSet(dpInbound.Tags).Exclude(mesh_proto.ServiceTag),
+			})
+		}
+	}
+	return ingresses
+}
+
+func GetAllDataplanes(resourceManager manager.ReadOnlyResourceManager) ([]*core_mesh.DataplaneResource, error) {
```

Comment:
Ok, I'm lost. How does ingress work with many meshes?
I see you are collecting dataplanes from all meshes but then you don't take it into account when building endpoint map